lib/cocoapods/native_target_ext.rb

```diff
@@ -0,0 +1,53 @@
+module Xcodeproj
+  class Project
+    module Object
+      class AbstractTarget
+        # Adds a dependency on the given metadata cache.
+        #
+        # @param  [MetadataCache] metadata
+        #         The metadata holding all the required metadata to construct itself as a target dependency.
+        #
+        # @return [void]
+        #
+        def add_cached_dependency(metadata)
+          unless dependency_for_cached_target(metadata)
```

Comment:
could remove the nesting with `return unless dependency_for_cached_target(metadata)`